### ISSUE TITLE
Gradio RAG chatbot demo 

### ISSUE THREAD

**Issue by yvrjsharma** [AkariAsai/self-rag#2]
Hi, congratulations, this is a very cool project, many kudos! Gradio would be happy to provide support for a locally linked as well as a Spaces-hosted demo.

Here are a few examples of Gradio demos from the community -
- [A Langchain demo from Howard Chase](https://huggingface.co/spaces/hwchase17/chat-langchain)
- A [Chat with pdf ](https://huggingface.co/spaces/fffiloni/langchain-chat-with-pdf) demo 

In case you're interested, this is a [step-by-step guide](https://huggingface.co/docs/hub/spaces-sdks-gradio) explaining the process of creating a gradio sdk on Spaces. 😊 This is our docs on [Community GPU Grants](https://huggingface.co/docs/hub/spaces-gpus#community-gpu-grants).

**Comment by AkariAsai** [comment]
Thank you so much for your suggestions. I would be super interested in that. At one point we were thinking about hosting the demo but was concerned about the costs... I'll try to file the community GPU grants.

**Comment by yvrjsharma** [comment]
Thanks for the response. If you have a demo on Spaces, can you raise a request for community grant and provide me with the link too?

**Comment by maxjeblick** [comment]
Would be great to have some gradio app and/or some end-to-end jupyter notebook example that extends the examples in the README. (E.g. similar to https://github.com/NielsRogge/Transformers-Tutorials examples)


`run...py` seem to be tailored towards the evaluation dataset (e.g. uses `args.task == "factscore":`, uses json as input-output) and is currently a bit hard to follow/use.
